Makefile

```diff
@@ -1,8 +1,14 @@
 ifneq ($O,)
-ifneq ($(shell echo $(O) | grep -- "/\.$$"),)
-$(error Cannot support output directory 'O' path ending with terminal subpath info "./")
-endif
-	out-dir := $O
+# Remove trailing / and /. in output path, rmdir does not like them
+out-dir := $(shell s=$O; while [ true ]; do \
+            [ "$$s" == "/" ] && break; \
+            s=`echo "$$s" | sed s\/\\\/$$\/\/ --`; \
+            [ "$$s" == "/" ] && break; \
+            s=`echo "$$s" | sed s\/\\\/\.$$\/\/ --`; \
+            echo $$s | grep --quiet /$$ && continue; \
+            echo $$s | grep --quiet /\.$$ && continue; \
+	    break; \
+        done; echo $$s)
```

Comment:
nice macro.

> (note: if O=/ or O=, $(out-dir) will be empty, do we need to handle those cases?)

if O is empty, out-dir is set through another sequence, so it's ok.
if O=/ (or fancy /././///./.), then yes, we must handle the case. I would propose to simply reject empty `$out-dir` contain. I think we don't expect to build straight in OS root directory.